google/cloud/aiplatform/training_jobs.py

```diff
@@ -15,9 +15,297 @@
 # limitations under the License.
 #
 
+import datetime
+import functools
+import pathlib
+import shutil
+import subprocess
+import sys
+import tempfile
+from typing import Callable, Dict, Optional, Sequence
 
-class TrainingJob:
-    pass
+
+from google.auth import credentials as auth_credentials
+from google.cloud.aiplatform import base
+from google.cloud.aiplatform import datasets
+from google.cloud.aiplatform import models
+from google.cloud.aiplatform_v1beta1 import PipelineServiceClient
+from google.cloud import storage
+
+
+def _timestamped_copy_to_gcs(
+    local_file_path: str,
+    staging_bucket: str,
+    project: Optional[str] = None,
+    credentials: Optional[auth_credentials.Credentials] = None,
+) -> str:
+    """Copies a local file to a GCS bucket.
+
+    The file copied to GCS is the name of the local file prepended with an
+    "aiplatform-{timestamp}-" string.
+
+    Args:
+        local_file_path (str): Required. Local file to copy to GCS.
+        staging_bucket (str):
+            Required. The staging bucket to copy to. (should not include gs://)
```

Comment:
I think in some other parts of the uCAIP API `gs://` is expected, perhaps we should keep this consistent.

Also the name `staging_bucket` suggests it should be a bucket, but in reality a GCS prefix will do just fine?

(example: https://github.com/googleapis/python-aiplatform/pull/13/files#diff-c3ab7043e19fe58fca6c927aad405f57839d6c40b651039d881afb8a7eba9040R29)

Reply:
Done.